### PR TITLE
The CRON job now is working and triggers builds on DockerHub

### PR DESCRIPTION
**Pull request by potiuk**
The CRON job from previous runs did not have everything working
after the emergency migration to Github Actions.

This change brings back following improvements:

* rebuilding images from the scratch in CRON job
* automatically upgrading all requirements to test if they are new
* pushing production images to github packages as cache
* pushing nightly tag to github

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by potiuk** [comment]
All green!

**Comment by zhongjiajie** [comment]
👍 LGTM, but curious why should we change schedule cron?

**Review by turbaszek — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -363,3 +368,31 @@ jobs:
...
+      - name: "Build CI image"
+        run: ./scripts/ci/ci_prepare_image_on_ci.sh
+      - name: "Tag commit"
+        run: |
+          git tag -f nightly-master HEAD
+      - name: Push tags
+        uses: ad-m/github-push-action@master
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          tags: true
```
Comment (kaxil) [inline .github/workflows/ci.yml:398]: Does this push to Master directly ??? If so, we shouldn't do that instead create a PR like the Dependency Bot

Comment (potiuk) [reply .github/workflows/ci.yml:398]: It only pushes nightly tag  - nothing else. I am adding an extra protection to make sure that this always  a good branch. But the whole idea is that this job pushes "nighty-master" tag after all the tests in nightly cron job are fine. I am also adding some comprehensive description of how the CI jobs work as different jobs have different purpose. Bear with me :)

**Comment by potiuk** [comment]
>  LGTM, but curious why should we change schedule cron?

Well. I am still experimenting with the best time of night to run it :).

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kaxil) [reply .github/workflows/ci.yml:398]: Sure

**Comment by zhongjiajie** [comment]
> > LGTM, but curious why should we change schedule cron?
> 
> Well. I am still experimenting with the best time of night to run it :).

I got it.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (potiuk) [reply .github/workflows/ci.yml:398]: There it goes @kaxil (and others) . Complete with description of the CI infrastructure we use and even a nice image showing the infrastructure.

**Pull request merged**